tests/e2e-instrumentation/instrumentation-java-multicontainer/chainsaw-test.yaml

```diff
@@ -40,12 +40,203 @@ spec:
     catch:
       - podLogs:
           selector: app=my-java-multi
+  - name: Make a request to the app
+    try:
+      - command:
+          env:
+          - name: appnamespace
+            value: ($namespace)
+          entrypoint: kubectl
+          args:
+            - get
+            - pod
+            - -n
+            - $appnamespace
+            - -l
+            - app=my-java-multi
+            - -o
+            - jsonpath={.items[0].metadata.name}
+          outputs:
+          - name: podName
+            value: ($stdout)
+      - proxy:
+          apiVersion: v1
+          kind: Pod
+          name: ($podName)
+          namespace: ($namespace)
+          port: "8080"
+  - name: Wait for telemetry data to be collected
```

Comment:
Do we need this step and especially the script?

Could the next step retry the metrics query od fail if it does not pass after some time?

Reply:
The script and the step is required for the time being, until this is implemented natively in Chainsaw. There is a feature request for that. https://github.com/kyverno/chainsaw/issues/1896 cc @eddycharly The other option is to add a sleep but I would prefer the script as we go onto the next step as soon as the strings are found in the pod logs. Otherwise we have to wait for whatever sleep duration that we have added. 

As for the metrics query step, this cannot be run in a loop for the required metrics to be found. The kubectl proxy is a single operation, the metrics assert works on the data fetched from the kubectl proxy. There is no retry at the try test step spec.